### PR TITLE
refactoring Query, fixed query errors

### PR DESCRIPTION
**Pull request by StigNorland**

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search2/common/Query.java [hunk 1]
```diff
@@ -90,7 +88,7 @@ protected Query() {
         searchParameters = new ConcurrentHashMap<>();
         pageParameters = new ConcurrentHashMap<>();
         otherRequiredKeys = new HashSet<>();
-        opensearchQueryTools = new QueryTools<>();
+        kQueryTools = new QueryTools<>();
```
Comment (truhacevkir) [inline search-commons/src/main/java/no/unit/nva/search2/common/Query.java:91]: Maybe just queryTools?

File: search-commons/src/main/java/no/unit/nva/search2/resource/ResourceQuery.java [hunk 2]
```diff
@@ -68,13 +90,28 @@
...
+            case FUNDING -> fundingQuery(key);
+            case CRISTIN_IDENTIFIER -> additionalIdentifierQuery(key, CRISTIN_AS_TYPE);
+            case SCOPUS_IDENTIFIER -> additionalIdentifierQuery(key, SCOPUS_AS_TYPE);
+            case HAS_PUBLIC_FILE -> publishedFileQuery(key);
+            case EXCLUDE_SUBUNITS -> excludeSubunitsQuery();
+            default -> {
+                logger.error("unhandled key -> {}", key.name());
+                yield Stream.empty();
+            }
+        };
```
Comment (truhacevkir) [inline search-commons/src/main/java/no/unit/nva/search2/resource/ResourceQuery.java:114]: Nice!

**Review by truhacevkir — APPROVED** [review]
Nice, but I would rename kQueryTools to queryTools. None knows what "k" stands for.

File: search-commons/src/main/java/no/unit/nva/search2/common/Query.java (hunk 1 above)
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/common/Query.java:91]: I had a good reason, but now I don't remember why

**Pull request merged**